packages/dds/cell/src/test/cell.spec.ts

```diff
@@ -161,55 +161,55 @@ describe("Cell", () => {
         });
 
         describe("Garbage Collection", () => {
-            let subCellCount = 0;
-            let expectedRoutes: string[] = [];
-
-            beforeEach(() => {
-                subCellCount = 0;
-                expectedRoutes = [];
-            });
-
-            // Return the remote SharedCell because we want to verify its summary data.
-            const getSharedObject = () => cell2;
-
-            async function addOutboundRoutes() {
-                const factory = new CellFactory();
-                const newSubCell = factory.create(dataStoreRuntime, `subCell-${++subCellCount}`);
-                cell.set(newSubCell.handle);
-                expectedRoutes = [ newSubCell.handle.absolutePath ];
-                containerRuntimeFactory.processAllMessages();
-            }
-
-            async function deleteOutboundRoutes() {
-                cell.delete();
-                expectedRoutes = [];
-                containerRuntimeFactory.processAllMessages();
+            class GCSharedCellProvider implements IGCTestProvider {
+                private subCellCount = 0;
+                private _expectedRoutes: string[] = [];
+
+                constructor() {
+                    this.subCellCount = 0;
+                    this._expectedRoutes = [];
+                }
+
+                public get sharedObject() {
+                    // Return the remote SharedCell because we want to verify its summary data.
+                    return cell2;
```

Comment:
Are you sure we want to reuse same object across multiple test instances? I.e. shouldn't it be member of this class, i.e. created and initialized every time instance of this class is created?

Reply:
Yeah, you are right. I will update it.

Reply:
Updated this as per my other comment.